nixos/modules/system/boot/systemd/journald.nix

```diff
@@ -127,13 +127,13 @@ in
         "systemd-journal-catalog-update.service"
         "systemd-journald-sync@.service"
       ]
-      ++ (lib.optional (!config.boot.isContainer) "systemd-journald-audit.socket")
+      ++ lib.optional (!config.boot.isContainer) "systemd-journald-audit.socket"
```

Comment:
I would prefer to just remove this check instead and include `systemd-journald-audit.socket` unconditionally

The unit has `ConditionSecurity=audit` so it will already automatically be skipped inside containers https://github.com/systemd/systemd/blob/7c087d802fa7ed68c7eb88b5b69bbdd6b93a86ae/units/systemd-journald-audit.socket#L15


and `wantedBy` is a soft dependency so there is no harm in always including it as far as I'm aware

Reply:
I do agree with @arianvp here on providing the full unit.
Got here because `systemd-analyze verify` complains about the partial unit installed on *boot.isContainer* machines in the current state (#360426).
I have tested this with a commit removing the optional here, and it works just fine in terms of `systemd-analyze verify`.
@arianvp Should I push that commit as a new PR?